--- a/configure
+++ b/configure
@@ -502,6 +502,7 @@ docker="no"
 # 2. Support --with/--without option
 default_devices="true"

+libargo="no"
 # 3. Automatically enable/disable other options
 tcg="enabled"
 cfi="false"
@@ -6036,6 +6037,16 @@ if test "$libudev" != "no" ; then
 if test "$meson" = git; then
     git_submodules="${git_submodules} meson"
 fi
+##########################################
+# argo support probe
+if test "$libargo" = "yes"; then
+    libargo_cflags=$($pkg_config --cflags libargo)
+    libargo_libs=$($pkg_config --libs libargo)
+    if ! compile_prog "$libargo_cflags" "$libargo_libs"; then
+        feature_not_found "libargo" "Install libargo devel"
+    fi
+fi
+

 case "$meson" in
     git | internal)
@@ -7417,6 +7429,11 @@ if test "$libudev" != "no"; then
   echo "CONFIG_RNG_NONE=y" >> $config_host_mak
 fi

+if test "$libargo" = "yes" ; then
+  echo "LIBARGO_CFLAGS=$libargo_cflags" >> $config_host_mak
+  echo "LIBARGO_LIBS=$libargo_libs" >> $config_host_mak
+fi
+
 # use included Linux headers
 if test "$linux" = "yes" ; then
   mkdir -p linux-headers
